File: leetcode/Easy/0026-remover-duplicatas-de-um-array-ordenado/resolucao_0026.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        writer = 0

        for pointer in range(1, len(nums)):
            if nums[pointer] != nums[pointer -1]:
                writer += 1
                nums[writer] = nums[pointer]

        return writer + 1

File: leetcode/Easy/0026-remover-duplicatas-de-um-array-ordenado/resolucao_0026.go
package main

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	writer := 0
	for pointer := 1; pointer < len(nums); pointer++ {
		if nums[pointer] != nums[pointer-1] {
			writer++
			nums[writer] = nums[pointer]
		}
	}

	return writer + 1
}

// remova o main ao testar no leetcode
func main() {}
